crates/bevy_reflect/src/array.rs

```diff
@@ -45,10 +45,8 @@ pub trait Array: Reflect {
 /// A container for compile-time array info.
 #[derive(Clone, Debug)]
 pub struct ArrayInfo {
-    type_name: &'static str,
-    type_id: TypeId,
-    item_type_name: &'static str,
-    item_type_id: TypeId,
+    type_value: ValueInfo,
+    item_type_value: ValueInfo,
```

Comment:
Bikeshed: I think these names are fine, but here are some others to consider:
1. `container_type`, `item_type`, `key_type`, `field_type`, etc.
2. `type_info`, `item_type_info`, `key_type_info`, `field_type_info`, etc.

For me, I think I'd lean more towards either keeping what you have now or going with Alternative 1 (2 is okay, but has the implication that it contains a `TypeInfo` not a `ValueInfo`).

I’m curious if you or anyone else has any thoughts on this?

Reply:
I just replaced `name` and `id` with `value` for consistency. I'm thinking maybe `key_type_value` and `item_type_value`.